ngx_http_vod_module.c

```diff
@@ -3512,6 +3516,11 @@ ngx_http_vod_http_reader_open_file(ngx_http_request_t* r, ngx_str_t* path, uint3
 	// Note: for http, no need to open any files, just save the remote uri
 	state->r = r;
 	state->cur_remote_suburi = *path;
+	state->upstream_location = ctx->submodule_context.conf->upstream_location;
+	if (ctx->state == STATE_MAP_OPEN)
```

Comment:
I think this is reversed, should use remote_upstream_location for the mp4, not the mapping. Also, please add an `else` to this if and move the previous line there (no reason to set and then override it)